### PR TITLE
send 'reloadRecommended' info to IDEs

### PR DESCRIPTION
**Pull request by devoncarew**
- send any reload hint message, and a reload hint ID, to IDEs after a reload
- use this to send a `reloadRecommended` hint to IDEs, in order to better support the `'Some program elements were changed during reload but did not run when the view was reassembled...`' message
- work towards https://github.com/flutter/flutter-intellij/issues/1398

@jcollins-g @tvolkert @rmacnak-google

### CODE REVIEW

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 1]
```diff
@@ -843,12 +843,25 @@ abstract class ResidentRunner {
...
   final int code;
+
+  /// A user facing message about the results of the operation.
   final String message;
+
+  /// An optional hint about the results of the operation. This is used to provide
+  /// sidecar data about the operation results. For example, this is used when
+  /// a reload is successful but some changed program elements where not run after a
+  /// reassemble.
   final String hint;
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/resident_runner.dart]: Maybe rename this `hintMessage` to make it clear that `hintId` is the key used by tools, whereas `hintMessage` is the user-facing message displayed to the user?

Comment (devoncarew) [reply packages/flutter_tools/lib/src/resident_runner.dart]: sgtm!

**Comment by devoncarew** [comment]
ptal  - I renamed `hint` to `hintMessage`

**Review by tvolkert — APPROVED** [review]
LGTM

**Pull request merged**